lib/nostrum/gateway/event.ex

```diff
@@ -0,0 +1,22 @@
+defmodule Nostrum.Gateway.Event do
+  alias Nostrum.Gateway.{HelloEvent, ReadyEvent}
+
+  @type t :: ReadyEvent.t()
+
+  @spec new(Enum.t()) :: {:ok, struct} | :error
+  def new(enum) do
+    string_map =
+      enum
+      |> Enum.map(fn {k, v} -> {to_string(k), v} end)
```

Comment:
Why are we converting to strings here and elsewhere throughout these modules?

Reply:
Did a benchmark just now on this. String comparison is too slow, so I'm going to change this to compare with atoms.